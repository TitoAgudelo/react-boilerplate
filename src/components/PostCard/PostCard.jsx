import React from "react";
import styled from "styled-components";

const PostCard = ({ post, comments }) => {
  return (
    <Container>
      <Title testId="postTitle">{post.title}</Title>
      <Body testId="postBody">{post.body}</Body>
      <SubTitle>Comments:</SubTitle>
      {comments.map((comment, index) => {
        if (comment.postId === post.id) {
          return (
            <CommentContainer key={`comment-${index}`}>
              <CommentTitle>{comment.body}</CommentTitle>
              <CommentText>By: {comment.name} - Email: {comment.email}</CommentText>
            </CommentContainer>
          )
        }
      })}
    </Container>
  )
}

const Container = styled.div`
  cursor: pointer;
  display: flex;
  flex-direction: column;
  width: 100%;
`;

const Title = styled.p`
  color: ${({ theme: { colors } }) => colors.red };
  font-family: 'Macho SemiBold';
  font-weight: bold;
  font-size: 16px;
  margin: 0;
  padding: 1rem 0 0 5px;
  text-decoration: none;
  text-transform: capitalize;
`;

const Body = styled.p`
  color: ${({ theme: { colors } }) => colors.blue };
  font-family: 'Macho Regular';
  font-size: 12px;
  margin: 0;
  margin-bottom: 1rem;
  padding: 5px;
  text-transform: none;
  text-transform: capitalize;
`;

const SubTitle = styled.p`
  color: ${({ theme: { colors } }) => colors.white };
  font-family: 'Macho SemiBold';
  font-size: 12px;
  margin: 0;
  padding: 5px;
  text-transform: none;
  text-transform: capitalize;
`;

const CommentContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 10px;;
`;

const CommentTitle = styled.p`
  color: ${({ theme: { colors } }) => colors.white };
  font-family: 'Macho SemiBold';
  font-weight: bold;
  font-size: 10px;
  margin: 0;
  padding: 5px;
  text-decoration: none;
  text-transform: capitalize;
`;

const CommentText = styled.p`
  color: ${({ theme: { colors } }) => colors.blue };
  font-family: 'Macho Regular';
  font-size: 8px;
  margin: 0;
  margin-bottom: .5rem;
  padding: 5px;
  text-transform: none;
  text-transform: capitalize;
`;

export default PostCard;
