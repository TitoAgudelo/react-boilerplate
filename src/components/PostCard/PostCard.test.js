import React from 'react';
import { shallow } from 'enzyme';
import PostCard from './PostCard';

let post = { 
  userId: 1,
  id: 1,
  title: "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
  body: "quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto"
};

let comments = [
  {
    postId: 1,
    id: 1,
    name: "id labore ex et quam laborum",
    email: "Eliseo@gardner.biz",
    body: "laudantium enim quasi est quidem magnam voluptate ipsam eos\ntempora quo necessitatibus\ndolor quam autem quasi\nreiciendis et nam sapiente accusantium"
  },
  {
    postId: 1,
    id: 2,
    name: "quo vero reiciendis velit similique earum",
    email: "Jayne_Kuhic@sydney.com",
    body: "est natus enim nihil est dolore omnis voluptatem numquam\net omnis occaecati quod ullam at\nvoluptatem error expedita pariatur\nnihil sint nostrum voluptatem reiciendis et"
  },
]

let wrapped = shallow(<PostCard post={post} comments={comments} />);
let title = "sunt aut facere repellat provident occaecati excepturi optio reprehenderit";
let body = "quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto";

describe('PostCard', () => {
  it('should render the PostCard Component correctly', () => {   
    expect(wrapped).toMatchSnapshot();
  });

  it('renders the Titles children', () => { 
    expect(wrapped.find('[testId="postTitle"]').text()).toEqual(title);
  });

  it('renders the body children', () => { 
    expect(wrapped.find('[testId="postBody"]').text()).toEqual(body);
  });
});
