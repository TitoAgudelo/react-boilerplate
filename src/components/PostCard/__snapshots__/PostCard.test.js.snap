// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard should render the PostCard Component correctly 1`] = `
<styled.div>
  <styled.p
    testId="postTitle"
  >
    sunt aut facere repellat provident occaecati excepturi optio reprehenderit
  </styled.p>
  <styled.p
    testId="postBody"
  >
    quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto
  </styled.p>
  <styled.p>
    Comments:
  </styled.p>
  <styled.div
    key="comment-0"
  >
    <styled.p>
      laudantium enim quasi est quidem magnam voluptate ipsam eos
tempora quo necessitatibus
dolor quam autem quasi
reiciendis et nam sapiente accusantium
    </styled.p>
    <styled.p>
      By: 
      id labore ex et quam laborum
       - Email: 
      Eliseo@gardner.biz
    </styled.p>
  </styled.div>
  <styled.div
    key="comment-1"
  >
    <styled.p>
      est natus enim nihil est dolore omnis voluptatem numquam
et omnis occaecati quod ullam at
voluptatem error expedita pariatur
nihil sint nostrum voluptatem reiciendis et
    </styled.p>
    <styled.p>
      By: 
      quo vero reiciendis velit similique earum
       - Email: 
      Jayne_Kuhic@sydney.com
    </styled.p>
  </styled.div>
</styled.div>
`;
